### PR TITLE
feat(code): Suggest experiment opportunities in setup discovery

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Problem

Setup currently only suggests bug fixes and cleanups. We want it to also suggest experiments to run.

## Changes

The setup discovery scan now also looks for places in the code where an A/B test would make sense (CTAs, pricing, signup flows, empty states) and shows them as suggestion cards alongside the bugs.

Behind the `posthog-code-experiment-suggestions` flag. Off in prod, on in dev.

If the user clicks an experiment card, the agent creates a feature flag, a draft experiment with a primary metric, and wires the variants into the code. It reuses the existing `run-experiment` skill prompt for that part.

The agent only includes an experiment card if it finds a real candidate. When it does, it drops a lower-priority bug/cleanup card to make room.

## How did you test this?

Built a small Vite + React + posthog-js app with a landing CTA, pricing, signup, empty state, and one XSS bug. Ran discovery on it. Got an experiment suggestion plus the XSS. Clicked the experiment card and the agent created the flag, the draft experiment (`signup_submitted` as primary metric, `pricing_plan_selected` as guardrail), and updated the code.

|Prompt|Output|
|----|----|
|image|image|

Follow-up: add analytics on what experiments are being suggested before we turn the flag on for more people.

## Publish to changelog?

no

### CODE REVIEW

File: apps/code/src/renderer/features/setup/prompts.ts [hunk 1]
```diff
@@ -21,7 +21,29 @@ Scan the codebase for issues in two tiers. Tier 1 applies to every repo. Tier 2
...
+
+If you find at least one credible Tier 3 experiment opportunity, include at least one experiment-category task in your output — even if doing so displaces a lower-impact Tier 1/2 finding. Do not fabricate an experiment to fill the slot: if no credible candidate exists, omit the category entirely.`;
+
+const BASE_ALLOWED_CATEGORIES =
+  "bug, security, dead_code, duplication, performance, stale_feature_flag, error_tracking, event_tracking, funnel";
+
+function buildDiscoveryRules(includeExperiments: boolean): string {
+  const allowed = includeExperiments
+    ? `${BASE_ALLOWED_CATEGORIES}, experiment`
+    : BASE_ALLOWED_CATEGORIES;
```
[the remark this answers is not in the document]
Comment (adboio) [reply apps/code/src/renderer/features/setup/prompts.ts]: grok is this true? 🤔

**Review by adboio — APPROVED** [review]
this is sick, thank you!! one minor greptile comment that seems legit but otherwise lgtm

File: apps/code/src/renderer/features/setup/prompts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply apps/code/src/renderer/features/setup/prompts.ts]: Fixed
